File: PlacementCellBackend/DTOs/CompanyInsights.cs
namespace PlacementCellBackend.DTOs
{
    public class CompanyInsights
    {
        public string CompanyId { get; set; } = string.Empty;
        public string CompanyName { get; set; } = string.Empty;
        public string Industry { get; set; } = string.Empty;
        public int TotalAlumniPlaced { get; set; }
        public int TotalFeedbacks { get; set; }
        public List<string> JobProfiles { get; set; } = new();
        public List<string> Locations { get; set; } = new();
        public decimal AverageCTC { get; set; }
    }

    public class CompanyRanking
    {
        public string CompanyId { get; set; } = string.Empty;
        public string CompanyName { get; set; } = string.Empty;
        public string Industry { get; set; } = string.Empty;
        public int TotalPlacements { get; set; }
        public int Rank { get; set; }
        public string AveragePackage { get; set; } = string.Empty;
        public List<YearlyPlacementCount> YearlyPlacements { get; set; } = new();
    }

    public class YearlyPlacementCount
    {
        public int Year { get; set; }
        public int Count { get; set; }
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/dtos/CompanyInsights.java
package com.example.placementicsbackend.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CompanyInsights {

    private String companyId = "";
    private String companyName = "";
    private String industry = "";
    private int totalAlumniPlaced;
    private int totalFeedbacks;
    private List<String> jobProfiles = new ArrayList<>();
    private List<String> locations = new ArrayList<>();
    private BigDecimal averageCTC = BigDecimal.ZERO;
}
